File: _Basic/windows-forms-introduction-with-visual-basic/7-windows-forms-introduction-with-visual-basic-m7-datasets-exercise-files/VB/Data/Data/AddProductForm.vb
Public Class AddProductForm
    Private _category As Category

    Public Sub New(category As Category)
        InitializeComponent()

        _category = Category
        CategoryTextBox.Text = _category.CategoryName
    End Sub

    Public ReadOnly Property Product As Product
        Get
            Dim result As New Product(0, NameTextBox.Text, _category.CategoryID, QuantityTextBox.Text, PriceTextBox.Text, StockTextBox.Text, OrderTextBox.Text, DiscontinuedCheckBox.Checked)
            Return result
        End Get
    End Property
End Class

File: _Basic/windows-forms-introduction-with-visual-basic/8-windows-forms-introduction-with-visual-basic-m8-linq-exercise-files/CS/Data/Data/AddProductForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Data
{
    public partial class AddProductForm : Form
    {
        private Category _category;

        public AddProductForm(Category category)
        {
            InitializeComponent();

            _category = category;
            CategoryTextBox.Text = _category.CategoryName;
        }

        public Product Product
        {
            get
            {
                var price = Decimal.Parse(PriceTextBox.Text);
                var stock = int.Parse(StockTextBox.Text);
                var onOrder = int.Parse(OrderTextBox.Text);

                var result = new Product(0, NameTextBox.Text, _category.CategoryID, QuantityTextBox.Text, price, stock, onOrder, DiscontinuedCheckBox.Checked);
                return result;
            }
        }
    }
}
